src/peft/tuners/tuners_utils.py

```diff
@@ -1068,6 +1077,14 @@ def _maybe_include_all_linear_layers(peft_config: PeftConfig, model: nn.Module)
                     module_names_to_exclude.add(last_module_name)
                     break
 
+    # we don't want nested LoRA layers, i.e. LoRA being applied to possibly existing lora_A, lora_B, etc.
+    # see 2390
+    for prefix, module in model.named_modules():
+        if isinstance(module, BaseTunerLayer):
+            for suffix, child in module.named_modules():
+                if suffix:
+                    module_names_to_exclude.add(f"{prefix}.{suffix}")
+
```

Comment:
If we subtract all the base tuner layers anyway, why gather the linear ones explicitly in the first place and not just extract all the base tuner layers?